### PR TITLE
Add subscription validation with Key Manager in the jwt authentication

### PR DESCRIPTION
**Pull request by tharindu1st**
Add subscription Validation with key manager

### CODE REVIEW

**Comment by tharindu1st** [comment]
Testcase https://github.com/wso2/product-apim/pull/7196

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/keys/WSAPIKeyDataStore.java [hunk 1]
```diff
@@ -115,6 +115,31 @@ public APIKeyValidationInfoDTO getAPIKeyData(String context, String apiVersion,
...
     }
 
+    @Override
+    public APIKeyValidationInfoDTO validateSubscription(String context, String version, String consumerKey)
+            throws APISecurityException {
+        APIKeyValidatorClient client = null;
+        try {
+            client = clientPool.get();
+            return client.validateSubscription(context, version, consumerKey);
+        }catch (APISecurityException ex) {
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/keys/WSAPIKeyDataStore.java]: Please format the code.

File: features/apimgt/org.wso2.carbon.apimgt.core.feature/src/main/resources/config/api-manager.xml [hunk 2]
```diff
@@ -743,4 +743,8 @@
         </LogTracer>
     </OpenTracer>
 
+    <JWTAuthentication>
+        <!--<EnableSubscriptionValidationViaKeyManager>false</EnableSubscriptionValidationViaKeyManager>-->
```
Comment (bhathiya) [inline features/apimgt/org.wso2.carbon.apimgt.core.feature/src/main/resources/config/api-manager.xml]: shall we uncomment this?

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java [hunk 3]
```diff
@@ -211,10 +223,35 @@ else if (RevokedJWTDataHolder.isJWTTokenSignatureExistsInRevokedMap(tokenSignatu
...
+                        throw new APISecurityException(apiKeyValidationInfoDTO.getValidationStatus(),
+                                "User is NOT authorized to access the Resource. API Subscription validation failed.");
+                    }
+                }
+                log.debug("Ignored subscription validation");
             }
 
             log.debug("JWT authentication successful.");
-            return GatewayUtils.generateAuthenticationContext(tokenSignature, payload, api, getApiLevelPolicy(), true);
+            return GatewayUtils.generateAuthenticationContext(tokenSignature, payload, api, null,  getApiLevelPolicy(), true);
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java]:
```suggestion
            return GatewayUtils.generateAuthenticationContext(tokenSignature, payload, api, null, getApiLevelPolicy(), true);
```

**Pull request merged**